sympy/external/tests/test_codegen.py

```diff
@@ -120,12 +120,12 @@ def try_run(commands):
     """Run a series of commands and only return True if all ran fine."""
     if pyodide_js:
         return False
-    null = open(os.devnull, 'w')
-    for command in commands:
-        retcode = subprocess.call(command, stdout=null, shell=True,
-                stderr=subprocess.STDOUT)
-        if retcode != 0:
-            return False
+    with open(os.devnull, 'w') as null:
```

Comment:
I found cases where `with` should be used to ensure that files are closed properly. That should probably be reviewed in bulk across the codebase.